src/yz_index_hashtree.erl

```diff
@@ -307,8 +307,7 @@ fold_keys(Partition, Tree) ->
                 insert(async, IndexN, BKey, Hash, Tree, [if_missing])
         end,
     Filter = [{partition, LogicalPartition}],
-    [yz_entropy:iterate_entropy_data(I, Filter, F) || I <- Indexes],
```

Comment:
So, fold_keys was crashing here when it couldn't get a response from the /entropy_data handler. Now, we iterate through but store an error atom if the latest iteration failed to give a response. Previously, if it crashed, we wouldn't get a `build_finished`... https://github.com/basho/yokozuna/pull/538/files#diff-9e0330457ea26c2e802f9524821758c6R558 ... call. Now, we throw a build_failed if the last iteration is an error (timeout) to requeue a poke in order to try and continue building the tree.

@bsparrow435 and I discussed the idea of only building a partial tree if errors/timeouts occur, but this could cause tons of exchanges if we're getting errors/timeouts that persist and occur early in the iteration cycle, as we'd *finish* building the tree without many of the keys. Something to continue thinking about though. 

/cc @fadushin @jonmeredith